### DISCUSSION TITLE
How to add new patterns to an existing project?

### DISCUSSION THREAD

**Discussion by happymimimix** [demberto/PyFLP#188] [Q&A]
I tried proj.patterns.append(), but it just give me this: 

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Patterns' object has no attribute 'append'

How am I supposed to add new patterns to my project? 
And also how to add new channels? 
And how to add note events in channels?

**Comment by happymimimix** [comment]
I'm trying to make a midi to flp converter with this. But got stuck at this step. 
Someone please help me out.

**Comment by demberto** [comment]
`Project.patterns` is an iterator, not a list. There's no way to create or add new patterns right now, you need to manually copy and append events. Take a look at the events patterns use and where they should be added by using FLPEdit.
